Sources/sourcekit-lsp/main.swift

```diff
@@ -18,134 +20,177 @@ import LSPLogging
+
+  @Option(
+    name: .customLong("Xcc", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all C compiler invocations"
+  )
+  var buildFlagsCc = [String]()
+
+  @Option(
+    name: .customLong("Xcxx", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all C++ compiler invocations"
+  )
+  var buildFlagsCxx = [String]()
+
+  @Option(
+    name: .customLong("Xlinker", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all linker invocations"
+  )
+  var buildFlagsLinker = [String]()
+
+  @Option(
+    name: .customLong("Xswiftc", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass flag through to all Swift compiler invocations"
+  )
+  var buildFlagsSwift = [String]()
+
+  @Option(
+    name: .customLong("Xclangd", withSingleDash: true),
+    parsing: .unconditionalSingleValue,
+    help: "Pass options to clangd command-line"
+  )
+  var clangdOptions = [String]()
+
+  @Option(
+    name: .customLong("index-store-path", withSingleDash: true),
+    help: "Override index-store-path from the build system"
+  )
+  var indexStorePath: AbsolutePath?
+
+  @Option(
+    name: .customLong("index-db-path", withSingleDash: true),
+    help: "Override index-database-path from the build system"
+  )
+  var indexDatabasePath: AbsolutePath?
+
+  @Option(
+    name: .customLong("completion-server-side-filtering", withSingleDash: true),
+    help: "Whether to enable server-side filtering in code-completion"
+  )
+  var completionServerSideFiltering = true
+
+  @Option(
+    name: .customLong("completion-max-results", withSingleDash: true),
+    help: "When server-side filtering is enabled, the maximum number of results to return"
+  )
+  var completionMaxResults = 200
+
+  func mapOptions() -> SourceKitServer.Options {
+    var serverOptions = SourceKitServer.Options()
+
+    serverOptions.buildSetup.configuration = buildConfiguration
+    serverOptions.buildSetup.path = buildPath
+    serverOptions.buildSetup.flags.cCompilerFlags = buildFlagsCc
+    serverOptions.buildSetup.flags.cxxCompilerFlags = buildFlagsCxx
+    serverOptions.buildSetup.flags.linkerFlags = buildFlagsLinker
+    serverOptions.buildSetup.flags.swiftCompilerFlags = buildFlagsSwift
+    serverOptions.clangdOptions = clangdOptions
+    serverOptions.indexOptions.indexStorePath = indexStorePath
+    serverOptions.indexOptions.indexDatabasePath = indexDatabasePath
+    serverOptions.completionOptions.serverSideFiltering = completionServerSideFiltering
+    serverOptions.completionOptions.maxResults = completionMaxResults
+
+    return serverOptions
   }
 
-  if let serverSideFiltering = parsedArguments.get(completionServerSideFiltering) {
-    result.serverOptions.completionOptions.serverSideFiltering = serverSideFiltering
+  func run() throws {
+    if !Logger.shared.setLogLevel(environmentVariable: "SOURCEKIT_LOGGING") {
```

Comment:
This inverts the behaviour - currently the command-line option is considered first.